gapis/api/vulkan/vulkan.go

```diff
@@ -66,8 +67,16 @@ func (*State) Root(ctx context.Context, p *path.State, r *path.ResolveConfig) (p
 
 // SetupInitialState recreates the command lamdas from the state block.
 // These are not encoded so we have to set them up here.
-func (s *State) SetupInitialState(ctx context.Context) {
+func (s *State) SetupInitialState(ctx context.Context, state *api.GlobalState) {
 	s.customState.init(s)
+
+	// Reserve memory for mapped ranges
+	for _, dm := range s.DeviceMemories().All() {
+		if uint64(dm.MappedLocation()) != uint64(0) {
+			state.ReserveMemory([]interval.U64Range{
```

Comment:
Not sure if its worth doing, but we could raise the interval array out of the loop, so we only have to allocate / expand internal range-list once.

Reply:
I don't expect there to be many. Can streamline it when we see an issue.